python/pyarrow/tests/test_scalars.py

```diff
@@ -238,13 +249,16 @@ def test_numerics():
     assert repr(s) == "<pyarrow.DoubleScalar: 1.5>"
     assert str(s) == "1.5"
     assert s.as_py() == 1.5
+    assert float(s) == 1.5
+    assert int(s) == 1
 
     # float16
-    s = pa.scalar(0.5, type='float16')
+    s = pa.scalar(np.float16(0.5), type='float16')
```

Comment:
Actually, NumPy is not needed for this anymore (thanks to the latest changes on git main):
```suggestion
    s = pa.scalar(0.5, type='float16')
```